src/syntax/parser.rs

```diff
@@ -833,22 +833,59 @@ fn args(p: &mut Parser) {
     p.wrap(m, SyntaxKind::Args);
 }
 
+// TODO (Marmare): remove success return value
+enum PatternKind {
+    Normal,
+    Destructuring,
+}
+
+fn pattern(p: &mut Parser) -> PatternKind {
+    let m = p.marker();
+    let unpacking = p.at(SyntaxKind::LeftParen);
+    if unpacking {
```

Comment:
I think its clear from the code that follows. Also, it would be nice if the terminology was clear: Either "unpacking" or "destructuring" (across all code, errors, and docs). I think I'm in favor of "destructuring".
```suggestion
    if p.at(SyntaxKind::LeftParen) {
```